bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,1222 @@
+        logger.info("Hayward polling cancelled");
+        super.dispose();
+    }
+
+    @Override
+    public void initialize() {
+        // Long running initialization should be done asynchronously in background.
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                "Opening connection to Hayward's Server");
+
+        // Load up the config and then get the connection to Hayward setup.
+        // HaywardConfig config = getConfigAs(HaywardConfig.class);
+        config = getConfigAs(HaywardConfig.class);
+
+        clearPolling(pollTelemetryFuture);
+        clearPolling(pollAlarmsFuture);
+
+        if (config.commandPollDelay == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DISABLED,
+                    "Command Poll Delay = 0.  Initialize binding disabled.");
+            return;
+        }
+
+        try {
+            if (!(login())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to Login to Hayward's server");
+                clearPolling(pollTelemetryFuture);
+                clearPolling(pollAlarmsFuture);
+                commFailureCount = 50;
+                initPolling(30);
+                return;
+            }
+
+            if (!(getSiteList())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to getMSP from Hayward's server");
+                clearPolling(pollTelemetryFuture);
+                clearPolling(pollAlarmsFuture);
+                commFailureCount = 50;
+                initPolling(30);
+                return;
+            }
+
+            if (!(getMspConfig())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unable to getConfig from Hayward's server");
+                clearPolling(pollTelemetryFuture);
+                clearPolling(pollAlarmsFuture);
+                commFailureCount = 50;
+                initPolling(30);
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.info("Succesfully opened connection to Hayward's server: {} Username:{}", config.hostname,
+                    config.username);
+
+            if (config.telemetryPollTime > 0) {
+                initPolling(0);
+                logger.info("Hayward Telemetry polling scheduled");
+            } else {
+                logger.info("Hayward Telemetry polling disabled");
+            }
+
+            if (config.alarmPollTime > 0) {
+                initAlarmPolling(1);
+                logger.info("Hayward Alarm polling scheduled");
+            } else {
+                logger.info("Hayward Alarm polling disabled");
+            }
+        } catch (Exception e) {
+            logger.error("Unable to open connection to Hayward's server: {} Username:{}", config.hostname,
```

Comment:
Any of the communications to the server can periodically timeout due to internet connection issues.  Rather than take the thing offline after a single connection failure, I wrote the polling such that it would retry up to 5 times before cancelling all polling jobs followed by calling the initialize routine which sets the bridge thing offline, dwells 30 seconds and starts over.

Reply:
Then, this should be debug.